### PR TITLE
Refactor CreateCaseCallbackService - use PaymentsProcessor

### PR DESCRIPTION
**Pull request by apereverzin**
### JIRA link (if applicable) ###
https://tools.hmcts.net/jira/browse/BPS-825

### Change description ###
Introduced PaymentsHandler

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[x] No
```

### CODE REVIEW

File: src/test/java/uk/gov/hmcts/reform/bulkscan/orchestrator/services/ccd/PaymentsHandlerTest.java [hunk 1]
```diff
@@ -0,0 +1,116 @@
...
+                .createCaseWith(builder -> builder
+                    .id(CASE_ID)
+                    .caseTypeId(CASE_TYPE_ID)
+                    .jurisdiction("some jurisdiction")
+                    .data(data)
+                );
+
+
+        // when
+        paymentsHandler.handlePayments(caseDetails,1L);
```
Comment (rkondratowicz) [inline src/test/java/uk/gov/hmcts/reform/bulkscan/orchestrator/services/ccd/PaymentsHandlerTest.java]: missing space

File: src/test/java/uk/gov/hmcts/reform/bulkscan/orchestrator/services/ccd/CreateCaseCallbackServiceTest.java [hunk 2]
```diff
@@ -658,16 +655,11 @@ void should_send_payment_message_when_case_has_payments() throws Exception {
...
-        assertThat(cmd.getValue().exceptionRecordRef).isEqualTo(Long.toString(CASE_ID));
-        assertThat(cmd.getValue().newCaseRef).isEqualTo(Long.toString(newCaseId));
-        assertThat(cmd.getValue().envelopeId).isEqualTo(envelopeId);
-        assertThat(cmd.getValue().jurisdiction).isEqualTo(jurisdiction);
+        verify(paymentsHandler).handlePayments(caseDetails, CASE_ID);
     }
 
     @Test
-    void should_not_send_payment_message_when_case_has_no_payments() throws Exception {
+    void should_call_payments_handler_when_case_has_no_payments() throws Exception {
```
Comment (rkondratowicz) [inline src/test/java/uk/gov/hmcts/reform/bulkscan/orchestrator/services/ccd/CreateCaseCallbackServiceTest.java]: I think those two tests can now be merged into one as it's always called.

File: src/main/java/uk/gov/hmcts/reform/bulkscan/orchestrator/services/ccd/PaymentsHandler.java [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+
+    private static final Logger log = LoggerFactory.getLogger(PaymentsHandler.class);
+
+    private final IPaymentsPublisher paymentsPublisher;
+
+    public PaymentsHandler(IPaymentsPublisher paymentsPublisher) {
+        this.paymentsPublisher = paymentsPublisher;
+    }
+
+    public void handlePayments(CaseDetails exceptionRecord, long newCaseId) {
```
Comment (rkondratowicz) [inline src/main/java/uk/gov/hmcts/reform/bulkscan/orchestrator/services/ccd/PaymentsHandler.java]: After extracting it from the 'context' of `CreateCaseCallbackService` this method name (and class name) sound **very** vague.

Comment (rkondratowicz) [reply src/main/java/uk/gov/hmcts/reform/bulkscan/orchestrator/services/ccd/PaymentsHandler.java]: will this class also handle creating the initial payment (when exception record is created)?

Comment (rkondratowicz) [reply src/main/java/uk/gov/hmcts/reform/bulkscan/orchestrator/services/ccd/PaymentsHandler.java]: Here we're essentially removing payments from exception record and adding them to newly created case.
So maybe something `movePayments` or `updatePayments`?

Comment (apereverzin) [reply src/main/java/uk/gov/hmcts/reform/bulkscan/orchestrator/services/ccd/PaymentsHandler.java]: movePaymentsToCase ?

Comment (apereverzin) [reply src/main/java/uk/gov/hmcts/reform/bulkscan/orchestrator/services/ccd/PaymentsHandler.java]: It actually sends UpdatePayments command to payments queue and is not in charge of any manipulations with payments so I think 'handleCasePayments' will be accurate enough

**Pull request merged**